politeiad/cache/cockroachdb/decred.go

```diff
@@ -45,6 +46,9 @@ type decred struct {
 	recordsdb *gorm.DB              // Database context
 	version   string                // Version of decred cache plugin
 	settings  []cache.PluginSetting // Plugin settings
+
+	bbMtx         sync.RWMutex
```

Comment:
You don't need to have a specific mutex for the best block yet.  It makes sense when you have multiple bits of shared state that are being updated simultaneously, but that's not the case here.  Just do:
```
type decred struct {
  sync.RWMutex
  ...
}
```

If in the future more stuff gets added that needs the lock held then you can break them out into individual mutexes.

Reply:
Makes sense mate, thanks for clearing this up. Fixed.